BaseClasses.py

```diff
@@ -1045,11 +1046,15 @@ def __init__(self, player: int, name: str = '', address: Optional[int] = None, p
         self.address = address
         self.parent_region = parent
 
-    def can_fill(self, state: CollectionState, item: Item, check_access=True) -> bool:
-        return ((self.always_allow(state, item) and item.name not in state.multiworld.worlds[item.player].options.non_local_items)
-                or ((self.progress_type != LocationProgressType.EXCLUDED or not (item.advancement or item.useful))
-                    and self.item_rule(item)
-                    and (not check_access or self.can_reach(state))))
+    def can_fill(self, state: CollectionState, item: Item, check_access: bool = True) -> bool:
+        return ((
```

Comment:
It feels like the way to clean this would be a few intermediate local variables. Formatting it is "better", I guess, but it's still not very readable at all

Reply:
With the number of times this function is called in generate, immediate local variables might add significant time.

I'm not saying it's not worth it - just that for this PR I'm avoiding deciding whether it's worth it.

Reply:
> immediate local variables might add significant time
Gosh I hate Python